src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer machineProducer = null;
+        if (type == Bulldozer.class) {
+            machineProducer = new BulldozerProducer();
+        } else if (type == Excavator.class) {
+            machineProducer = new ExcavatorProducer();
+        } else if (type == Truck.class) {
+            machineProducer = new TruckProducer();
+        }
+        List<? extends Machine> machines = machineProducer.get();
```

Comment:
we will have NPE if the input class is not a Bulldozer or Excavator or Truck

Reply:
done